website/docs/r/compute_disk.html.markdown

```diff
@@ -34,69 +66,176 @@ resource "google_compute_disk" "default" {
+  sourceSnapshot parameter, or specify it alone to create an empty
+  persistent disk.
+
+  If you specify this field along with sourceImage or sourceSnapshot,
+  the value of sizeGb must not be less than the size of the sourceImage
+  or the size of the snapshot.
+* `image` -
+  (Optional)
+  The image from which to initialize this disk. This can be
+  one of: the image's `self_link`, `projects/{project}/global/images/{image}`,
+  `projects/{project}/global/images/family/{family}`, `global/images/{image}`,
+  `global/images/family/{family}`, `family/{family}`, `{project}/{family}`,
+  `{project}/{image}`, `{family}`, or `{image}`. If referred by family, the
+  images names must include the family name. If they don't, use the
+  [google_compute_image data source](/docs/providers/google/d/datasource_compute_image.html).
+  For instance, the image `centos-6-v20180104` includes its family name `centos-6`.
+  These images can be referred by family name here.
+* `type` -
+  (Optional)
+  URL of the disk type resource describing which disk type to use to
+  create the disk. Provide this when creating the disk.
+* `disk_encryption_key` -
+  (Optional)
+  Encrypts the disk using a customer-supplied encryption key.
+
+  After you encrypt a disk with a customer-supplied key, you must
+  provide the same key if you use the disk later (e.g. to create a disk
+  snapshot or an image, or to attach the disk to a virtual machine).
+
+  Customer-supplied encryption keys do not protect access to metadata of
+  the disk.
+
+  If you do not provide an encryption key when creating the disk, then
+  the disk will be encrypted using an automatically generated key and
+  you do not need to provide a key to use the disk later.  Structure is documented below.
+* `source_image_encryption_key` -
+  (Optional)
+  The customer-supplied encryption key of the source image. Required if
+  the source image is protected by a customer-supplied encryption key.  Structure is documented below.
+* `snapshot` -
+  (Optional)
+  The source snapshot used to create this disk. You can provide this as
+  a partial or full URL to the resource. For example, the following are
+  valid values:
+
+  * https://www.googleapis.com/compute/v1/projects/project/global/
+        snapshots/snapshot
+  * projects/project/global/snapshots/snapshot
+  * global/snapshots/snapshot
+  * snapshot
+* `source_snapshot_encryption_key` -
+  (Optional)
+  The customer-supplied encryption key of the source snapshot. Required
+  if the source snapshot is protected by a customer-supplied encryption
+  key.  Structure is documented below.
+* `project` (Optional) The ID of the project in which the resource belongs.
+    If it is not provided, the provider project is used.
+
+The `disk_encryption_key` block supports:
+* `raw_key` -
+  (Optional)
+  Specifies a 256-bit customer-supplied encryption key, encoded in
+  RFC 4648 base64 to either encrypt or decrypt this resource.
+* `sha256` -
```

Comment:
Hmmm, should this be in the `Attributes Reference` section?

Reply:
I thought this was the right place for it.  Do you mean because it's computed?  Hm.  That introduces a challenge for generating the docs, but of course it's doable.